packages/react/package.json

```diff
@@ -119,6 +127,41 @@
       "@commitlint/config-conventional"
     ]
   },
+  "nyc": {
+    "report-dir": "cypress/coverage",
+    "reporter": [
+      "html",
+      "text-summary",
+      "lcov",
+      "json"
+    ],
+    "include": [
+      "src/components/**/*.js?(x)"
+    ],
+    "exclude": [
+      "src/**/*.story.js?(x)",
+      "src/**/hooks/*.js",
+      "src/components/SuiteHeader/util/suiteHeaderData.js",
+      "src/components/SuiteHeader/util/uiresources.js",
+      "src/components/FileUploader/stories/*.jsx",
+      "src/components/Table/AsyncTable/*.js?(x)",
+      "src/components/WizardInline/**/*.js?(x)",
+      "src/components/Page/(EditPage|PageHero).jsx",
+      "src/components/Dashboard/(Dashboard|CardRenderer).jsx",
+      "src/components/MapCard/storyFiles/*.jsx",
+      "src/components/BarChartCard/barChartUtils.js",
+      "src/components/ComboChartCard/comboChartHelpers.js",
+      "src/components/DashboardEditor/DashboardEditor.jsx",
+      "src/components/Table/Table.test.helpers.js",
+      "src/components/Table/TablePropTypes.js",
+      "src/components/Table/baseTableReducer.js",
+      "src/components/Table/tableReducer.js"
+    ],
+    "branches": 80,
```

Comment:
So NYC is the threshold that will run on our coveralls reporting. NYC is the tool that takes coverage from both and combines them into a single report. I had thought about removing the jest config globals but was unsure if we should leave them in. Since most unit test will be done with Jest it may be helpful to have your run fail when your jest test are not above the threshold  so you do not have to run the combine script every time. 

So moving forward you would write your code. Write your test. If you have maxed out the code that can be covered in Jest you can then write an exception for the jest threshold and move to writing your cypress test. Not sure if this the best flow. Definitely open to hearing your thoughts on the dev experience and what your preference would be.

Reply:
Hmmm...This is a tricky one. Maybe we remove individual thresholds from jest/cypress and then add another script that checks the total coverage from the merged version similar to the `coverage:check` script in [Part 3 of this article](https://dev.to/muratkeremozcan/combined-unit-e2e-code-coverage-case-study-on-a-real-life-system-using-angular-jest-cypress-gitlab-35nk).

Reply:
That is what we are doing already aside from not having jest thresholds removed. Our combine script has the flags `--check-coverage --per-file`. Which will check against the threshold numbers after it combines the reports. 

The issue is that if you are not writing and cypress test and you do not have the coverage generated for them you will have to run the cypress  test just to get the combined report. Not ideal. With jest thresholds in place they can at least be checked for the majority of cases without having to run any additional code.

Reply:
Oh, I'm following now. If we move these checks into a `pre-push` hook instead of a `pre-commit` hook like I [mention below](https://github.com/carbon-design-system/carbon-addons-iot-react/pull/2464#discussion_r670620018) then I don't see much of an issue having to run both. It gets more out of the way of general development and commits, but forces the coverage before pushes, so you run everything once when attempting to push and then get the feedback to make whatever changes. I think that improves the DX pretty well over the current setup.

Reply:
Setting testing on a pre-push hook.